src/router.rs

```diff
@@ -113,23 +146,14 @@ impl<Data: Clone + Send + Sync + 'static> Router<Data> {
         &'a self,
         path: &'a str,
         method: &http::Method,
-    ) -> Option<RouteResult<'a, Data>> {
-        let (route, route_match) = self.table.route(path)?;
-        // If it is a HTTP HEAD request then check if there is a callback in the endpoints map
-        // if not then fallback to the behavior of HTTP GET else proceed as usual
-        let endpoint =
-            if method == http::Method::HEAD && !route.endpoints.contains_key(&http::Method::HEAD) {
-                route.endpoints.get(&http::Method::GET)?
-            } else {
-                route.endpoints.get(method)?
-            };
-        let middleware = &*route.middleware;
-
-        Some(RouteResult {
-            endpoint,
-            params: route_match,
-            middleware,
-        })
+        default_handler: &'a Arc<BoxedEndpoint<Data>>,
+    ) -> RouteResult<'a, Data> {
+        match self.table.route(path) {
+            Some((route, route_match)) => route_match_success(route, route_match, method)
+                .or_else(|| Some(route_match_failure(default_handler, &self.middleware_base)))
+                .unwrap(),
```

Comment:
Oops, missed this; this can be written with one `unwrap_or_else` call

Reply:
fixed!